src/npcmove.cpp

```diff
@@ -2017,6 +2000,93 @@ void npc::drop_items(int weight, int volume)
     update_worst_item_value();
 }
 
+bool npc::find_corpse_to_pulp()
+{
+    if( is_following() && !rules.allow_pulp ) {
+        return false;
+    }
+
+    // Pathing with overdraw can get expensive, limit it
+    int path_counter = 4;
+    const auto check_tile = [this, &path_counter]( const tripoint &p ) -> const item * {
+        if( !g->m.sees_some_items( p, *this ) || !sees( p ) ) {
+            return nullptr;
+        }
+
+        const auto items = g->m.i_at( p );
+        const item * found = nullptr;
+        for( const item &it : items ) {
+            // Pulp only stuff that revives, but don't pulp acid stuff
+            // That is, if you aren't protected from this stuff!
+            if( it.can_revive() && !is_dangerous_field( it.get_mtype()->bloodType() ) ) {
+                found = &it;
+            }
+        }
```

Comment:
The loop will continue iterating even after finding a suitable corpse to pulp. It could either stop at once (update_path and return from here) or check them all and reject tiles with dangerous ones to overcome the first limitation.